dartsim/src/SimulationFeatures.cc

```diff
@@ -52,7 +59,54 @@ void SimulationFeatures::WorldForwardStep(
 
   // TODO(MXG): Parse input
   world->step();
-  // TODO(MXG): Fill in output and state
+  this->WriteRequiredData(_h);
+  this->Write(_h.Get<JointPositions>());
+  // TODO(MXG): Fill in state
+}
+
+void SimulationFeatures::Write(JointPositions &/*_positions*/) const
+{
+  // TODO(adlarkin) implement this, if it's needed?
+}
```

Comment:
This was also inspired by the [DARTDoublePendulum test](https://github.com/ignitionrobotics/ign-physics/blob/c77ac49b01e7a57cd0a6f6ba76c46837c65210ab/test/plugins/DARTDoublePendulum.cc#L273). However, I don't think we need to do anything with joints here since we only need to know what links moved (https://github.com/ignitionrobotics/ign-physics/issues/219). If this is true, I'll remove this method.

Reply:
Yeah, I would suggest we remove this for now instead of leaving it empty.

Reply:
removed in https://github.com/ignitionrobotics/ign-physics/pull/223/commits/ad12778a0af2f312222f108db4bd90ebf22502cb